homeassistant/components/dsmr/config_flow.py

```diff
@@ -47,6 +47,8 @@ def __init__(self, host: str | None, port: int, dsmr_version: str) -> None:
         self._equipment_identifier = obis_ref.EQUIPMENT_IDENTIFIER
         if dsmr_version == "5L":
             self._equipment_identifier = obis_ref.LUXEMBOURG_EQUIPMENT_IDENTIFIER
+        if dsmr_version == "Q3D":
+            self._equipment_identifier = obis_ref.Q3D_EQUIPMENT_IDENTIFIER
```

Comment:
Home Assistant requires 100% coverage on the configuration flows, please add a test for this added case.